android-sdk/src/androidTest/java/com/example/android_sdk/PaymentFormPresentedEventTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.android_sdk
+
+import android.content.Context
+import android.os.Build
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class PaymentFormPresentedEventTest {
+    private var appContext: Context? = null
+
+    @Before
+    fun setUp() {
+        appContext = InstrumentationRegistry.getInstrumentation().targetContext
+    }
+
+    @Test
+    fun test_language_property_of_paymentFormPresented_event_returnsCorrectValue() {
+        setLocale(Locale.GERMANY)
+        assertTrue("de" == Locale.getDefault().language)
+    }
```

Comment:
this test is checking the output of `Locale.getDefault().language`, but we wanted unit tests for the properties sent in the logs. Can we please rework the log event so that it can be tested instead?

We should also be testing country as well as language.

Reply:
Sure. This is a configuration test for device language change. For this, I have added this instrumented test case. Additionally, I have added unit test for the properties that we are sending in the event.